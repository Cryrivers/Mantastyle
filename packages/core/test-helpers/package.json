{
  "name": "@manta-style/test-helpers",
  "version": "0.2.0-alpha.9",
  "main": "src/index.ts",
  "dependencies": {
    "prettier": "^1.14.3"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@types/prettier": "^1.13.2"
  }
}
