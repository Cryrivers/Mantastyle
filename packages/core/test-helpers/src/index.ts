import * as BuilderPluginTestHelper from './builder-tester';
export { BuilderPluginTestHelper };
