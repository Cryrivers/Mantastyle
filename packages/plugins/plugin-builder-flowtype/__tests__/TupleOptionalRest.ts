import getTranspiledString from '../src/utils/transpile';
import { BuilderPluginTestHelper } from '@manta-style/test-helpers';

const testTranspiledString = BuilderPluginTestHelper.testTranspiledString(
  getTranspiledString,
);

describe('Tuple, Optional and Rest', () => {
  test('Tuple', () => {
    expect(
      testTranspiledString('type Test = [number, number]'),
    ).toMatchSnapshot();
  });
  test('Optional', () => {
    expect(
      testTranspiledString('type Test = [number, number, ?number]'),
    ).toMatchSnapshot();
  });
  xtest('Rest', () => {
    expect(
      testTranspiledString('type Test = [number, number, string]'),
    ).toMatchSnapshot();
  });
});
