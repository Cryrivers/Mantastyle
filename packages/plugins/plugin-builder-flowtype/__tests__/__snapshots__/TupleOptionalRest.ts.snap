// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuple, Optional and Rest Optional 1`] = `
"FROM:
type Test = [number, number, ?number]

--------------------------------------------------------
TO:
'use strict';

var _runtime = _interopRequireDefault(require('@manta-style/runtime'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

var Test = _runtime.default.TypeAliasDeclaration(
  'Test',
  function(typeFactory) {
    var type = _runtime.default.TupleType([
      _runtime.default.NumberKeyword,
      _runtime.default.NumberKeyword,
      _runtime.default.OptionalType(_runtime.default.NumberKeyword),
    ]);

    return type;
  },
  [],
);
"
`;

exports[`Tuple, Optional and Rest Tuple 1`] = `
"FROM:
type Test = [number, number]

--------------------------------------------------------
TO:
'use strict';

var _runtime = _interopRequireDefault(require('@manta-style/runtime'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

var Test = _runtime.default.TypeAliasDeclaration(
  'Test',
  function(typeFactory) {
    var type = _runtime.default.TupleType([
      _runtime.default.NumberKeyword,
      _runtime.default.NumberKeyword,
    ]);

    return type;
  },
  [],
);
"
`;
